libsel4/sel4_arch_include/aarch64/sel4/sel4_arch/deprecated.h

```diff
@@ -6,4 +6,40 @@
 
 #pragma once
 
-/* nothing here */
+#define seL4_ARM_PageDirectory seL4_ARM_PageTable;
+#define seL4_ARM_PageDirectory_Map seL4_ARM_PageTable_Map
+#define seL4_ARM_PageDirectory_Unmap seL4_ARM_PageTable_Unmap
+#define ARMPageDirectoryMap ARMPageTableMap
+#define ARMPageDirectoryUnmap ARMPageTableUnmap
+
+#define seL4_PageDirBits seL4_PageTableBits
+#define seL4_PageDirEntryBits seL4_PageTableEntryBits
+#define seL4_PageDirIndexBits seL4_PageTableIndexBits
+#define seL4_ARM_PageDirectoryObject seL4_ARM_PageTableObject
+#define seL4_PUDEntryBits 3
+
+#if defined(CONFIG_ARM_HYPERVISOR_SUPPORT) && defined (CONFIG_ARM_PA_SIZE_BITS_40)
+
+#define seL4_PGDBits 0
+#define seL4_PGDEntryBits 0
+#define seL4_PGDIndexBits    0
+
+#define seL4_PUDBits 13
+#define seL4_PUDIndexBits 10
+#define seL4_ARM_PageUpperDirectoryObject seL4_ARM_VSpaceObject
+
+#else
+
+#define seL4_PGDBits 12
+#define seL4_PGDEntryBits 3
+#define seL4_PGDIndexBits    9
+
+#define seL4_PUDBits 12
+#define seL4_PUDIndexBits 9
+#define seL4_ARM_PageGlobalDirectoryObject seL4_ARM_VSpaceObject
```

Comment:
The build failure in `libsel4vka` looks like `seL4_ARM_PageGlobalDirectoryObject` was declared as a name even in hyp configs with `CONFIG_ARM_PA_SIZE_BITS_40` defined. Not sure if it is better to do that again so less code breaks, or if it's better to break that code, because it shouldn't have been using that name in the first place.

Reply:
I added a fix to my PR in seL4_libs for this change